src/averageAges.js

```diff
@@ -55,7 +81,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people
+    .filter(kid => people.some(mother => kid.mother === mother.name));
+
+  const sons = children.filter(son => son.sex === 'm');
+
+  const ageDifferentChildren = children
+    .map(kid => kid.born - people
+      .find(mother => mother.name === kid.mother).born);
+
+  const ageDifferentSons = sons
+    .map(kid => kid.born - people
+      .find(mother => mother.name === kid.mother).born);
```

Comment:
you don't need this two variables at the same time, you only need one or another, so maybe let's combine this logic into one variable